stl/inc/cmath

```diff
@@ -561,28 +605,43 @@ _STD _Common_float_type_t<_Ty1, _Ty2> remquo(_Ty1 _Left, _Ty2 _Right, int* _Pquo
 #endif // _HAS_IF_CONSTEXPR
 }
 
-#define _GENERIC_MATH1R(FUN, RET)                                            \
+#define _GENERIC_MATH1_BASE(NAME, RET, FUN)                                  \
     template <class _Ty, _STD enable_if_t<_STD is_integral_v<_Ty>, int> = 0> \
-    _NODISCARD RET FUN(_Ty _Left) noexcept /* strengthened */ {              \
-        return _CSTD FUN(static_cast<double>(_Left));                        \
+    _NODISCARD RET NAME(_Ty _Left) noexcept /* strengthened */ {             \
+        return FUN(static_cast<double>(_Left));                              \
     }
 
-#define _GENERIC_MATH1(FUN) _GENERIC_MATH1R(FUN, double)
+#define _GENERIC_MATH1R(FUN, RET) _GENERIC_MATH1_BASE(FUN, RET, _CSTD FUN)
+#define _GENERIC_MATH1(FUN)       _GENERIC_MATH1R(FUN, double)
+
+#ifdef __clang__
+#define _GENERIC_MATH1I(FUN, INTRIN) _GENERIC_MATH1R(FUN, double)
+#else // ^^^ __clang__ / !__clang__ vvv
+#define _GENERIC_MATH1I(FUN, INTRIN) _GENERIC_MATH1_BASE(FUN, double, INTRIN)

```

Comment:
Since all of the callers match the argument pattern `meow`, `__meow`, we could simply omit the second parameter:
```suggestion
#define _GENERIC_MATH1I(FUN) _GENERIC_MATH1R(FUN, double)
#else // ^^^ __clang__ / !__clang__ vvv
#define _GENERIC_MATH1I(FUN) _GENERIC_MATH1_BASE(FUN, double, __ ## FUN)
```

Reply:
I initially made it like that because it's not true in the case of `signbit` (the intrinsic in that case is `__signbitvalue`), but #519 is blocking us from adding `signbit` as well to the set of integer overloads.

Reply:
I also believe that being explicit also makes it clearer to people who dare attempt to modify `<cmath>` in the future

Reply:
I think that the explicit second parameter is fine, due to `__signbitvalue` in the future as you mentioned. It also makes the code more searchable (synthesizing identifiers via token-pasting makes it impossible to search for the intrinsic names).